src/sqliteformat.cpp

```diff
@@ -48,9 +47,13 @@ using namespace mKCal;
 class mKCal::SqliteFormat::Private
 {
 public:
-    Private(SqliteStorage *storage, sqlite3 *database)
-        : mStorage(storage), mDatabase(database)
+    Private(SqliteFormat *format, sqlite3 *database, const QTimeZone &timeZone)
+        : mFormat(format), mDatabase(database)
+        , mOriginTime(QDateTime(QDate(1970, 1, 1), QTime(0, 0, 0), Qt::UTC))
```

Comment:
Not related entirely to these changes but I've been pondering if there's some point in this custom epoch compared to just using QDateTime::toSecsSinceEpoch() / fromSecsSinceEpoch().

Reply:
I think the point of having a date hard-coded here (and choosing epoch is not a necessity) is because date times in the database are stored as integers corresponding to a shift in seconds to a reference. So one needs to define this reference. What is surprising to me is that this reference is hard-coded instead of being stored in the database. Because at the moment, any change for another reference in the source code will break all existing databases...

But since this reference date time is hard-coded and set to epoch, it could be simpler to actually use the Qt routines converting from epoch directly and drop this definition. I'm going to try...